### PR TITLE
Update the install docs

### PR DESCRIPTION
**Pull request by angeloskath**
Add a small paragraph about installing from PyPI.

### CODE REVIEW

File: docs/src/install.rst [hunk 1]
```diff
@@ -1,14 +1,30 @@
...
+
 MLX Data consists of a C++ library with shallow python bindings and some Python
 helpers. You can choose to build and use only the C++ library or simply pip
 install the Python library which transparently builds the required C++ backend.
 
 We plan to provide prebuilt binaries for easy install in the future.
 
 Dependencies
-------------
+^^^^^^^^^^^^
```
Comment (awni) [inline docs/src/install.rst:27]: Won't some of the dependencies build automatically with the cmake?

Comment (awni) [reply docs/src/install.rst:27]: In fact, aws is the only one that doesn't? If I understand correctly, we should simplify the paragraphs below to just have a line on using aws.

Comment (angeloskath) [reply docs/src/install.rst:26]: None except if one is using the super build. I still think it is valid for people to dynamically link if they are building on their own. It is much faster and allows building with many things missing.

I did fix the docs though. I had forgotten to update the old part of the installation. Let me know if it needs a few more clarifications.

**Review by awni — APPROVED** [review]
LGTM, thanks!

**Pull request merged**